src/equation_systems/equation_system.C

```diff
@@ -106,12 +106,53 @@ EquationSystem::ApplyBoundaryConditions(platypus::BCMap & bc_map)
         test_var_name, _lfs.GetRef(test_var_name), _test_pfespaces.at(i)->GetParMesh());
   }
 }
+
 void
 EquationSystem::FormLinearSystem(mfem::OperatorHandle & op,
                                  mfem::BlockVector & trueX,
                                  mfem::BlockVector & trueRHS)
 {
 
+  switch (_assembly_level)
+  {
+    case mfem::AssemblyLevel::LEGACY:
+      FormLegacySystem(op, trueX, trueRHS);
+      break;
+    default:
+      MFEM_VERIFY(_test_var_names.size() == 1,
+                  "Non-legacy assembly is only supported for single-variable systems");
+      MFEM_VERIFY(_mblfs.size() != 0, "Non-legacy assembly is only supported for square systems");
```

Comment:
Should this check be `_mblfs.size() == 0`? Since you're checking that there are no mixed bilinear forms in the system?

Reply:
Yeah, this is looking like a typo. I'm just confused as to why it has passed the tests

Reply:
Fixed!